pkg/modules/authorization.go

```diff
@@ -631,6 +633,543 @@ func AuthorizationServerDeployment(ctx context.Context, isDeleting bool, op util
 		return err
 	}
 
+	err = applyDeleteAuthorizationStorageService(ctx, isDeleting, cr, ctrlClient)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// AuthorizationStorageService - apply/delete storage service deployment and volume objects
+func applyDeleteAuthorizationStorageService(ctx context.Context, isDeleting bool, cr csmv1.ContainerStorageModule, ctrlClient crclient.Client) error {
+	authModule, err := getAuthorizationModule(cr)
+	if err != nil {
+		return err
+	}
+
+	switch semver.Major(authModule.ConfigVersion) {
+	case "v2":
+		return authorizationStorageServiceV2(ctx, isDeleting, cr, ctrlClient)
+	case "v1":
+		return authorizationStorageServiceV1(ctx, isDeleting, cr, ctrlClient)
+	default:
+		return fmt.Errorf("authorization major version %s not supported", semver.Major(authModule.ConfigVersion))
+	}
+}
+
+func authorizationStorageServiceV1(ctx context.Context, isDeleting bool, cr csmv1.ContainerStorageModule, ctrlClient crclient.Client) error {
+	authModule, err := getAuthorizationModule(cr)
+	if err != nil {
+		return err
+	}
+
+	// get component variables
+	image := ""
+	for _, component := range authModule.Components {
+		switch component.Name {
+		case AuthProxyServerComponent:
+			image = component.StorageService
+		}
+	}
+
+	replicas := int32(1)
+	// set deployment
+	deployment := appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
```

Comment:
Missing TypeMeta for deployment

Reply:
Fixed.

Reply:
Manually deployed Authorization to verify change.